controller/src/main/java/cern/jarrace/controller/rest/controller/AgentContainerController.java

```diff
@@ -0,0 +1,104 @@
+        System.out.println("Deployed " + name);
+        String path = writeFile(name, jar);
+        paths.put(name, path);
+        entrypoints.put(name, new ArrayList<>());
+        List<String> args = new ArrayList<>();
+        args.add("cern.jarrace.agent.AgentContainer");
+        args.add(name);
+        args.add("localhost:8080");
+        startContainer(path, args);
+    }
+
+    @RequestMapping(value = "/service/register/{name}", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void registerService(@PathVariable(value = "name") String name, @RequestBody Entrypoint entrypoint){
+
+        if(entrypoint.getAgentName() == null || entrypoint.getAgentName().isEmpty()) {
+            throw new IllegalArgumentException("Agent name must be different than null and not empty");
+        }
+        if(entrypoint.getPath() == null || entrypoint.getPath().isEmpty()) {
+            throw new IllegalArgumentException("Path must be different than null and not empty");
+        }
+        entrypoints.get(name).add(entrypoint);
+        LOGGER.info(entrypoints.toString());
+    }
+
+    @RequestMapping(value = "/container/list", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public Set<String> listContainers() {
+        return entrypoints.keySet();
+    }
+
+    @RequestMapping(value = "/{name}/entrypoint/list", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public List<Entrypoint> listServices(@PathVariable(value = "name") String containerName) {
+        List<Entrypoint> toReturn = this.entrypoints.get(containerName);
+        if (toReturn == null) {
+            toReturn = new ArrayList<>();
+        }
+        return toReturn;
+    }
+
+    @RequestMapping(value = "/{name}/{entrypoint}/start", method = RequestMethod.GET)
+    public String runService(@PathVariable("name") String name, @PathVariable int entrypoint) throws IOException {
+        List<Entrypoint> entries = entrypoints.get(name);
+        Entrypoint entryPoint = entries.get(entrypoint);
+        List<String> args = new ArrayList<>();
+        args.add("cern.jarrace.agent.AgentRunner");
+        args.add(entryPoint.getAgentName());
+        args.add(entryPoint.getPath());
+        startContainer(paths.get(name), args);
+        return null;
+    }
+
+    private void startContainer(String path, List<String> args) throws IOException {
+        List<String> command = new ArrayList<>();
+        command.add(String.format("%s/bin/java", System.getProperty("java.home")));
+        command.add("-cp");
+        command.add(path);
+        command.addAll(args);
+
+        LOGGER.info(String.format("Starting agent container [%s]", command.toString()));
+        ProcessBuilder processBuilder = new ProcessBuilder(command);
+        processBuilder.inheritIO().start();
+    }
+
+    private String writeFile(String name, byte[] jar) throws IOException {
```

Comment:
What does the user (human or application) receive if an IOException actually occurs (e.g. no more disk space, no permissions, etc.)?